lib/rouge/lexers/idris.rb

```diff
@@ -0,0 +1,206 @@
+        negate not null or pi pred print product putChar putStr putStrLn
+        readFile recip repeat replicate return reverse scanl scanl1 sequence
+        sequence_ show sin sinh snd span splitAt sqrt succ sum tail take
+        takeWhile tan tanh uncurry unlines unwords unzip unzip3 words
+        writeFile zip zip3 zipWith zipWith3
+      )
+
+      state :basic do
+        rule %r/\s+/m, Text
+        rule %r/{-#/, Comment::Preproc, :comment_preproc
+        rule %r/{-/, Comment::Multiline, :comment
+        rule %r/^\|\|\|.*$/, Comment::Doc
+        rule %r/--(?![!#\$\%&*+.\/<=>?@\^\|_~]).*?$/, Comment::Single
+      end
+
+      # nested commenting
+      state :comment do
+        rule %r/-}/, Comment::Multiline, :pop!
+        rule %r/{-/, Comment::Multiline, :comment
+        rule %r/[^-{}]+/, Comment::Multiline
+        rule %r/[-{}]/, Comment::Multiline
+      end
+      state :comment_preproc do
+        rule %r/-}/, Comment::Preproc, :pop!
+        rule %r/{-/, Comment::Preproc, :comment
+        rule %r/[^-{}]+/, Comment::Preproc
+        rule %r/[-{}]/, Comment::Preproc
+      end
+
+      state :directive do
+        rule %r/\%(default)\s+(total|partial)/, Keyword  # totality
+        rule %r/\%(access)\s+(public|abstract|private|export)/, Keyword  # export
+        rule %r/\%(language)\s+(.*)/, Keyword  # language
+        rule %r/\%(provide)\s+.*\s+(with)\s+/, Keyword  # type
+      end
+
+      state :prelude do
+        rule %r/\b(Type|Exists|World|IO|IntTy|FTy|File|Mode|Dec|Bool|Ordering|Either|IsJust|List|Maybe|Nat|Stream|StrM|Not|Lazy|Inf)\s/, Keyword::Type
+        rule %r/\b(Eq|Ord|Num|MinBound|MaxBound|Integral|Applicative|Alternative|Cast|Foldable|Functor|Monad|Traversable|Uninhabited|Semigroup|Monoid)\s/, Name::Class
+        rule %r/\b(?:#{prelude_function.join('|')})[ ]+(?![=:-])/, Name::Builtin
+      end
+
+      state :root do
+        mixin :basic
+        mixin :directive
+
+        rule %r/\bimport\b/, Keyword::Reserved, :import
+        rule %r/\bmodule\b/, Keyword::Reserved, :module
+        rule %r/\b(?:#{reserved.join('|')})\b/, Keyword::Reserved
+        rule %r/\b(Just|Nothing|Left|Right|True|False|LT|LTE|EQ|GT|GTE)\b/, Keyword::Constant
+        # function signature
+        rule %r/^([\w']+)\s*(:)/, Name::Function
+        # should be below as you can override names defined in Prelude
+        mixin :prelude
+        rule %r/[_a-z][\w']*/, Name
+        rule %r/[A-Z][\w']*/, Keyword::Type
+
+        # lambda operator
+        rule %r(\\(?![:!#\$\%&*+.\\/<=>?@^\|~-]+)), Name::Function
+        # special operators
+        rule %r((<-|::|->|=>|=)(?![:!#\$\%&*+.\\/<=>?@^\|~-]+)), Operator
+        # constructor/type operators
+        rule %r(:[:!#\$\%&*+.\\/<=>?@^\|~-]*), Operator
+        # other operators
+        rule %r([:!#\$\%&*+.\\/<=>?@^\|~-]+), Operator
+
+        rule %r/\d+e[+-]?\d+/i, Num::Float
+        rule %r/\d+\.\d+(e[+-]?\d+)?/i, Num::Float
```

Comment:
Could you combine these rules like this?

```suggestion
        rule %r/\d+(\.\d+)?(e[+-]?\d+)?/i, Num::Float
```

Reply:
Addressed in https://github.com/rouge-ruby/rouge/pull/1464/commits/5e638d46c828f3eb3ed01202ca5f52a79c583d34